### ISSUE TITLE
Special characters in folder names aren't properly treated

### ISSUE THREAD

**Issue by pdobrescu** [PressLabs/gitfs#185]
When a folder (and maybe a file too) contains special characters, its parent folder cannot be listed.

E.g. the folder `abc` contains the sub-folder called `ghí`. Because of the `í`, running `ls` on `abc` returns:

`ls: reading directory .: Bad address`

**Comment by vtemian** [comment]
Can't reproduce. Can you give another example?

**Comment by pdobrescu** [comment]
Nope, this was the only case. Just to make it clear: one of the folders in the themes sub-folder was named `ghí`. And because of that I couldn't list or browse the themes folder.

**Comment by vtemian** [comment]
Try to monkey test gitfs with different sftp clients

**Comment by vtemian** [comment]
```13:29:56 Error:             mkdir /dobrestii/current/wp-content/root/dslkf sdfkjs fsdkj/Another ghí folder with spaces this time and a very very long name that has a big number of very weird characters. Yes we can have such customers :) ®†`˙©∆∫µ˜/blogit-wp/inc/nc-options/options/validation: failure```

**Comment by pdobrescu** [comment]
I managed to crash it using FileZilla. Here's the FileZilla log to check the operations I've done:

https://gist.github.com/pdobrescu/f3c7c2949ce632e3b054

**Comment by Blank000** [comment]
I am totally new to github. Can anyone help me that how can I contribute to open sources. I read a lot of articles, pages but didn't understand well.
Thank you.....

**Comment by leahlang4d2** [comment]
Hey blank000 did you figure it out? Here is a good [resource (https://help.github.com/articles/creating-a-pull-request-from-a-fork/)! @Blank000
